### ISSUE TITLE
[JENKINS-4753] gcc warnings not parsed correctly

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#2269]
<p>We're seeing frequent reports from the warnings plugin about "gcc error" which<br/>
are definitely not errors, but rather warnings that originate from use of<br/>
templates ands cause many lines of output.<br/>
Here's one example of such warning(s):</p>

<p>warnings.cc: In function 'void print(T) <span class="error">&#91;with T = Test&#93;</span>':<br/>
warnings.cc:9:   instantiated from 'void out(T) <span class="error">&#91;with T = Test&#93;</span>'<br/>
warnings.cc:14:   instantiated from here<br/>
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'long unsigned int'<br/>
warnings.cc:6: warning:   in call to 'std::basic_ostream<_CharT, _Traits>&<br/>
std::basic_ostream<_CharT, _Traits>::operator<<(int) [with _CharT = char,<br/>
_Traits = std::char_traits<char>]'<br/>
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'long int'<br/>
warnings.cc:6: warning:   in call to 'std::basic_ostream<_CharT, _Traits>&<br/>
std::basic_ostream<_CharT, _Traits>::operator<<(int) [with _CharT = char,<br/>
_Traits = std::char_traits<char>]'<br/>
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'unsigned int'<br/>
warnings.cc:6: warning:   in call to 'std::basic_ostream<_CharT, _Traits>&<br/>
std::basic_ostream<_CharT, _Traits>::operator<<(int) [with _CharT = char,<br/>
_Traits = std::char_traits<char>]'</p>

<p>From this I'm told there is 1 gcc error (in line 9) and 4 warnings in line 6 (3x<br/>
"Passing 'Test'" and 1 "in call to").<br/>
Theoretically, it might be just one warning, but 3 warnings would be ok - the 3<br/>
"passing 'Test'" warnings, preferably annotated/joined with the other lines to<br/>
make things clearer. And the error is not an error, it's just information.<br/>
It's loosely related to issue 4260 (which seems to eat the "instantiated from<br/>
here" message) and issue 4712 (about "note:" in output).</p>

<p>To reproduce: I'll attach warnings.cc. Compile with "g++ -Wsign-promo".</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=torbent">torbent</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753" target="_blank">gcc warnings not parsed correctly</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=fchateau">fchateau</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2010-03-25T10:47:53+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We're seeing frequent reports from the warnings plugin about "gcc error" which
are definitely not errors, but rather warnings that originate from use of
templates ands cause many lines of output.
Here's one example of such warning(s):</p>

<p>warnings.cc: In function 'void print(T) <span class="error">&#91;with T = Test&#93;</span>':
warnings.cc:9:   instantiated from 'void out(T) <span class="error">&#91;with T = Test&#93;</span>'
warnings.cc:14:   instantiated from here
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'long unsigned int'
warnings.cc:6: warning:   in call to 'std::basic_ostream&lt;_CharT, _Traits&gt;&amp;
std::basic_ostream&lt;_CharT, _Traits&gt;::operator&lt;&lt;(int) [with _CharT = char,
_Traits = std::char_traits&lt;char&gt;]'
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'long int'
warnings.cc:6: warning:   in call to 'std::basic_ostream&lt;_CharT, _Traits&gt;&amp;
std::basic_ostream&lt;_CharT, _Traits&gt;::operator&lt;&lt;(int) [with _CharT = char,
_Traits = std::char_traits&lt;char&gt;]'
warnings.cc:6: warning: passing 'Test' chooses 'int' over 'unsigned int'
warnings.cc:6: warning:   in call to 'std::basic_ostream&lt;_CharT, _Traits&gt;&amp;
std::basic_ostream&lt;_CharT, _Traits&gt;::operator&lt;&lt;(int) [with _CharT = char,
_Traits = std::char_traits&lt;char&gt;]'</p>

<p>From this I'm told there is 1 gcc error (in line 9) and 4 warnings in line 6 (3x
"Passing 'Test'" and 1 "in call to").
Theoretically, it might be just one warning, but 3 warnings would be ok - the 3
"passing 'Test'" warnings, preferably annotated/joined with the other lines to
make things clearer. And the error is not an error, it's just information.
It's loosely related to issue 4260 (which seems to eat the "instantiated from
here" message) and issue 4712 (about "note:" in output).</p>

<p>To reproduce: I'll attach warnings.cc. Compile with "g++ -Wsign-promo".</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>
<details><summary><i>1 attachment</i></summary>

- [warnings.cc](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/18873/warnings.cc)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=133272&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-133272">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created an attachment (id=980)
Source code to reproduce. Compile with g++ -Wsign-promo</p>
</pre>
</details>

</ul>
</details>

<p>Created an attachment (id=980)<br/>
Source code to reproduce. Compile with g++ -Wsign-promo</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=133273&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-133273">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The issue concerns the 'gcc' parser.
If I switch to the parser 'gcc-include', all warnings disappear. As far as I can
tell, that's also a defect.</p>
</pre>
</details>

</ul>
</details>

<p>The issue concerns the 'gcc' parser.<br/>
If I switch to the parser 'gcc-include', all warnings disappear. As far as I can<br/>
tell, that's also a defect.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=133274&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-133274">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hmm, this issue and <a href="https://issues.jenkins.io/browse/JENKINS-4260" title="False Warnings reported when using the exclude" class="issue-link" data-issue-key="JENKINS-4260"><del>JENKINS-4260</del></a> seems to be not easy to solve. Maybe there
should be a new parser that uses several lines as context. Currently the gcc
parser is implemented as a single line parser.</p>
</pre>
</details>

</ul>
</details>

<p>Hmm, this issue and <a href="https://issue-redirect.jenkins.io/issue/4260" title="False Warnings reported when using the exclude" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/4260">JENKINS-4260</a>"><del>JENKINS-4260</del></a> seems to be not easy to solve. Maybe there<br/>
should be a new parser that uses several lines as context. Currently the gcc<br/>
parser is implemented as a single line parser.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>rolfcampbell:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=134013&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-134013">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm seeing this problems as well for any line in the form "filename: information".</p>

<p>AFAIK: gcc errors are only in the form "filename: error: information".  Why not keep it as a one-line parser and just use that format?</p>
</pre>
</details>

</ul>
</details>

<p>I'm seeing this problems as well for any line in the form "filename: information".</p>

<p>AFAIK: gcc errors are only in the form "filename: error: information".  Why not keep it as a one-line parser and just use that format?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=134024&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-134024">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>rolfcampbell, I don't think I understand your suggestion ... ?
Anyway, this is a /warnings/ parser so errors are kinda unimportant, IMHO. And gcc /will/ output multiline messages (be it info, warnings, errors) so a singleline parser probably isn't the correct tool for the job. I haven't looked at the code, but a multiline parser shouldn't be all that difficult, should it?</p>
</pre>
</details>

</ul>
</details>

<p>rolfcampbell, I don't think I understand your suggestion ... ?<br/>
Anyway, this is a /warnings/ parser so errors are kinda unimportant, IMHO. And gcc /will/ output multiline messages (be it info, warnings, errors) so a singleline parser probably isn't the correct tool for the job. I haven't looked at the code, but a multiline parser shouldn't be all that difficult, should it?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=134105&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-134105">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Maybe the default warning type of this parser should not be error<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>I think the parsing is not so easy, since currently there are several different output formats modeled in the GCC parser (one line message, linker message, multi-line includes). Maybe the parser should be split into several parsers instead of trying to use one parser for every message type.</p>
</pre>
</details>

</ul>
</details>

<p>Maybe the default warning type of this parser should not be error<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>I think the parsing is not so easy, since currently there are several different output formats modeled in the GCC parser (one line message, linker message, multi-line includes). Maybe the parser should be split into several parsers instead of trying to use one parser for every message type.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=134403&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-134403">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Someone interested in improving the GCC parser? I don't work with gcc so I actually don't know much about the possible warning variations... </p>
</pre>
</details>

</ul>
</details>

<p>Someone interested in improving the GCC parser? I don't work with gcc so I actually don't know much about the possible warning variations... </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fchateau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=135859&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135859">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I think that if you want to correctly parse and retrieve template errors and contextual informations, it will take more than a regular expression parser. You will have to define a full blown grammar and do not restrict to single lines.
Currently it seems all parsers rely on regular expressions. Fortunately the base interface WarningsParser do not, so it should be possible to implement a GrammarParser over it without re-architecturing things.</p>

<p>However I think it won't be easy to find the correct grammar. Error output can get rather complex and I can't find any documentation about it on gcc website.
Maybe there are folks around there that already developed something close. I was thinking to Eclipse CDT people who must probably do GCC output parsing to show list of errors.
I gave a quick look at their code, and found some grammar based parsers, but I am not sure it was really parsing gcc errors because it didn't find any reference to "warning" or "error" keywords in the grammar, and the syntax just didn't look like to match the shape of gcc errors. Anyway I'm pretty sure there must be something worth of interest there.</p>
</pre>
</details>

</ul>
</details>

<p>I think that if you want to correctly parse and retrieve template errors and contextual informations, it will take more than a regular expression parser. You will have to define a full blown grammar and do not restrict to single lines.<br/>
Currently it seems all parsers rely on regular expressions. Fortunately the base interface WarningsParser do not, so it should be possible to implement a GrammarParser over it without re-architecturing things.</p>

<p>However I think it won't be easy to find the correct grammar. Error output can get rather complex and I can't find any documentation about it on gcc website.<br/>
Maybe there are folks around there that already developed something close. I was thinking to Eclipse CDT people who must probably do GCC output parsing to show list of errors.<br/>
I gave a quick look at their code, and found some grammar based parsers, but I am not sure it was really parsing gcc errors because it didn't find any reference to "warning" or "error" keywords in the grammar, and the syntax just didn't look like to match the shape of gcc errors. Anyway I'm pretty sure there must be something worth of interest there.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fchateau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=135879&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135879">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>What version of g++ do you use ? I suspect a 3.x version, because using g++ 4.x (tried with x=1,2,3,4), I get a different message:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">warnings.cc: In function <span class="code-quote">'void print(T) [with T = Test]'</span>:
warnings.cc:9:   instantiated from <span class="code-quote">'void out(T) [with T = Test]'</span>
warnings.cc:14:   instantiated from here
warnings.cc:6: warning: passing <span class="code-quote">'Test'</span> chooses <span class="code-quote">'<span class="code-object">int</span>'</span> over <span class="code-quote">'unsigned <span class="code-object">int</span>'</span>
warnings.cc:6: warning:   in call to <span class="code-quote">'std::basic_ostream&lt;_CharT, _Traits&gt;&amp; std::basic_ostream&lt;_CharT, _Traits&gt;::<span class="code-keyword">operator</span>&lt;&lt;(<span class="code-object">int</span>) [with _CharT = <span class="code-object">char</span>, _Traits = std::char_traits&lt;<span class="code-object">char</span>&gt;]'</span>
</pre>
</div></div>

<p>I think error output has changed a lot between gcc 3 and 4 and I wonder if it would not be better to create separate parsers.
I managed to write a regular expression giving better results, while still using single line approach, but it relied on the fact that relevant compiler messages (which excludes contextual informations like: "instanciated from", "In file include from", "in call to", "from ...", etc...) uses "warning", "error" and "note" keywords. It seems to be the case for g++ 4.x messages but it was not systematic in earlier versions.</p>

<p>Here is my regexp for compiler messages:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">^([^\n]+)(?::(\d+))?(?::\d+:)?: (warning|error): (.*)$
</pre>
</div></div>
<p>path=group1, line=group2, severity=group3, message=group4
I do volontarily ignore messages with "note" severity because in all cases I have been confronted to recently, they only gives additional informations about the previous error, and I'm pretty sure that they don't stand for a another compilation problem on their own.</p>

<p>I also wrote a regexp for linker messages:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">^([^\n]+)(?::(\d+))?(?:\(.*\+0x\d+\))?: ((?:undefined reference to|could not read symbols: ).*)|.*/ld: (.*)$
</pre>
</div></div>
<p>path=group1, line=group2, message=group3, ld_message=group4
This one fixes a few missing matches in the current regexp. It misses messages like:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">foo.cpp:(.text+0x15): undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
</pre>
</div></div>

<p>which is what you get when you have a missing library and debug infos are turned off. Otherwise, with debug turned on, you would have line information, like this:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">/tmp/cctDpmmr.o: In function `main':
/home/me/dev/temp/foo.cpp:7: undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
</pre>
</div></div>

<p>I think there must be more messages than: "undefined reference to" and "could not read symbols", but for now this is all I was able to generate, and I can't find any documentation about this matter.
Unfortunately we have to list messages explicitely otherwise will get compiler and not relevant messages also.</p>
</pre>
</details>

</ul>
</details>

<p>What version of g++ do you use ? I suspect a 3.x version, because using g++ 4.x (tried with x=1,2,3,4), I get a different message:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">warnings.cc: In function <span class="code-quote">'void print(T) [with T = Test]'</span>:
warnings.cc:9:   instantiated from <span class="code-quote">'void out(T) [with T = Test]'</span>
warnings.cc:14:   instantiated from here
warnings.cc:6: warning: passing <span class="code-quote">'Test'</span> chooses <span class="code-quote">'<span class="code-object">int</span>'</span> over <span class="code-quote">'unsigned <span class="code-object">int</span>'</span>
warnings.cc:6: warning:   in call to <span class="code-quote">'std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::<span class="code-keyword">operator</span><<(<span class="code-object">int</span>) [with _CharT = <span class="code-object">char</span>, _Traits = std::char_traits<<span class="code-object">char</span>>]'</span>
</pre>
</div></div>

<p>I think error output has changed a lot between gcc 3 and 4 and I wonder if it would not be better to create separate parsers.<br/>
I managed to write a regular expression giving better results, while still using single line approach, but it relied on the fact that relevant compiler messages (which excludes contextual informations like: "instanciated from", "In file include from", "in call to", "from ...", etc...) uses "warning", "error" and "note" keywords. It seems to be the case for g++ 4.x messages but it was not systematic in earlier versions.</p>

<p>Here is my regexp for compiler messages:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">^([^\n]+)(?::(\d+))?(?::\d+:)?: (warning|error): (.*)$
</pre>
</div></div>
<p>path=group1, line=group2, severity=group3, message=group4<br/>
I do volontarily ignore messages with "note" severity because in all cases I have been confronted to recently, they only gives additional informations about the previous error, and I'm pretty sure that they don't stand for a another compilation problem on their own.</p>

<p>I also wrote a regexp for linker messages:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">^([^\n]+)(?::(\d+))?(?:\(.*\+0x\d+\))?: ((?:undefined reference to|could not read symbols: ).*)|.*/ld: (.*)$
</pre>
</div></div>
<p>path=group1, line=group2, message=group3, ld_message=group4<br/>
This one fixes a few missing matches in the current regexp. It misses messages like:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">foo.cpp:(.text+0x15): undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
</pre>
</div></div>

<p>which is what you get when you have a missing library and debug infos are turned off. Otherwise, with debug turned on, you would have line information, like this:</p>
<div class="code panel" style="border-style: solid;border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">/tmp/cctDpmmr.o: In function `main':
/home/me/dev/temp/foo.cpp:7: undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
</pre>
</div></div>

<p>I think there must be more messages than: "undefined reference to" and "could not read symbols", but for now this is all I was able to generate, and I can't find any documentation about this matter.<br/>
Unfortunately we have to list messages explicitely otherwise will get compiler and not relevant messages also.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=135880&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135880">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I would also suggest to write individual parsers for different message types (gcc3, gcc4, linker, etc.). Otherwise the regular expression will be bloated (see <a href="https://issues.jenkins.io/browse/JENKINS-5185" title="Improve performance of parsers for long log files" class="issue-link" data-issue-key="JENKINS-5185"><del>JENKINS-5185</del></a>) and there will be a lot of false positives.</p>

<p>Are you interested in providing a (or some<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> new parsers?</p>
</pre>
</details>

</ul>
</details>

<p>I would also suggest to write individual parsers for different message types (gcc3, gcc4, linker, etc.). Otherwise the regular expression will be bloated (see <a href="https://issue-redirect.jenkins.io/issue/5185" title="Improve performance of parsers for long log files" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/5185">JENKINS-5185</a>"><del>JENKINS-5185</del></a>) and there will be a lot of false positives.</p>

<p>Are you interested in providing a (or some<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> new parsers?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fchateau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=135883&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135883">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, I would be interested in implementing a new parser, because I need it quick.
First I would like to create a fully working g++ 4.x error parsing.
At term, I would be also interested in developing error parsers for other tools we use in our project:</p>
<ul class="alternate" type="square">
	<li>make</li>
	<li>Qt tools (qmake, uic, moc)</li>
	<li>doxygen</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Yes, I would be interested in implementing a new parser, because I need it quick.<br/>
First I would like to create a fully working g++ 4.x error parsing.<br/>
At term, I would be also interested in developing error parsers for other tools we use in our project:</p>
<ul class="alternate" type="square">
	<li>make</li>
	<li>Qt tools (qmake, uic, moc)</li>
	<li>doxygen</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fchateau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=135885&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-135885">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I have news about Eclipse CDT GCC error parser. I looked thouroughly into the source code and I found how GCC messages are parsed.
Actually, parsing is done with a generic Regexp parser dynamically configured with an XML file and a properties file.
There are several regular expressions involved in parsing GCC errors.</p>

<p>If you're interested see CDT Sources at: :pserver:anonymous@dev.eclipse.org:2401/cvsroot/tools
in the directory: org.eclipse.cdt/all/org.eclipse.cdt.core
files: plugin.xml plugin.properties</p>

<p>Here are the regular expressions:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">CDTGNUCErrorParser.name=CDT GNU C/C++ Error Parser
CDTGNUCErrorParser.regex.ReportedOnlyOnce=(.*?):(\\d+):(\\d+:)? .*\\(Each undeclared identifier is reported only once.*
CDTGNUCErrorParser.regex.ForEachFunctionItAppearsIn=(.*?):(\\d+):(\\d+:)? .*<span class="code-keyword">for</span> each function it appears in.\\).*
CDTGNUCErrorParser.regex.ReportedOnlyOncePerInputFile=(.*?):(\\d+):(\\d+:)? .*<span class="code-keyword">this</span> will be reported only once per input file.*
CDTGNUCErrorParser.regex.InstantiatedFromHere=(.*?):(\\d+):(\\d+:)?\\s*(.*instantiated from here.*)
CDTGNUCErrorParser.regex.GenericInfo=(.*?):(\\d+):(\\d+:)?\\s*(([Nn]ote)|(NOTE)|([Ii]nfo)|(INFO)): (.*)
CDTGNUCErrorParser.regex.ParseErrorBefore=(.*?):(\\d+):(\\d+:)? (parse error before.*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.ErrorUndeclared=(.*?):(\\d+):(\\d+:)? [Ee]rror: ([`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] undeclared .*)
CDTGNUCErrorParser.regex.ErrorConflictingTypesFor=(.*?):(\\d+):(\\d+:)? [Ee]rror: (conflicting types <span class="code-keyword">for</span> .*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.GenericError=(.*?):(\\d+):(\\d+:)?\\s*(([Ee]rror)|(ERROR)): (.*)
CDTGNUCErrorParser.regex.DefinedButNotUsed=(.*?):(\\d+):(\\d+:)? [Ww]arning: ([`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] defined but not used.*)
CDTGNUCErrorParser.regex.WarningConflictingTypesFor=(.*?):(\\d+):(\\d+:)? [Ww]arning: (conflicting types <span class="code-keyword">for</span> .*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.GenericWarning=(.*?):(\\d+):(\\d+:)?\\s*(([Ww]arning)|(WARNING)): (.*)
CDTGNUCErrorParser.regex.OtherError=(.*?):(\\d+):(\\d+:)? (.*)

CDTGNULinkerErrorParser.name=CDT GNU Linker Error Parser
CDTGNULinkerErrorParser.regex.InFunction=(.*?):?(\\(\\.\\w+\\+.*\\))?:\\s*(In function [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>]:)
CDTGNULinkerErrorParser.regex.WarningDangerousFunction=(.*?):(\\d+):(\\d+:)? ([Ww]arning:)?\\s*(the use of [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] is dangerous, better use [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNULinkerErrorParser.regex.TextWarning=(.*?):?\\(\\.\\w+\\+.*\\): [Ww]arning:? (.*)
CDTGNULinkerErrorParser.regex.TextError=(.*?):?\\(\\.\\w+\\+.*\\): (.*)
CDTGNULinkerErrorParser.regex.ldWarning=ld(\\.exe)?: [Ww]arning:? (.*)
CDTGNULinkerErrorParser.regex.ldError=ld(\\.exe)?: (.*)
</pre>
</div></div>
<p>And here is the piece of XML where they are declared:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">   &lt;extension
         id=<span class="code-quote">"GCCErrorParser"</span>
         name=<span class="code-quote">"%CDTGNUCErrorParser.name"</span>
         point=<span class="code-quote">"org.eclipse.cdt.core.ErrorParser"</span>&gt;
      &lt;errorparser
            class=<span class="code-quote">"org.eclipse.cdt.core.errorparsers.RegexErrorParser"</span>
            id=<span class="code-quote">"org.eclipse.cdt.core.GCCErrorParser"</span>
            name=<span class="code-quote">"%CDTGNUCErrorParser.name"</span>&gt;
         &lt;pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ReportedOnlyOnce<span class="code-quote">" severity="</span>Ignore"/&gt;
         &lt;pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ForEachFunctionItAppearsIn<span class="code-quote">" severity="</span>Ignore"/&gt;
         &lt;pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ReportedOnlyOncePerInputFile<span class="code-quote">" severity="</span>Ignore"/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ErrorUndeclared"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ErrorConflictingTypesFor"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ParseErrorBefore"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.DefinedButNotUsed"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$5"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.WarningConflictingTypesFor"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$5"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$5"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNULinkerErrorParser.regex.WarningDangerousFunction"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$6"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.InstantiatedFromHere"</span> severity=<span class="code-quote">"Info"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$7"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericError"</span> severity=<span class="code-quote">"Error"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$7"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericWarning"</span> severity=<span class="code-quote">"Warning"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$9"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericInfo"</span> severity=<span class="code-quote">"Info"</span>/&gt;
         &lt;pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.OtherError"</span> severity=<span class="code-quote">"Error"</span>/&gt;
      &lt;/errorparser&gt;
   &lt;/extension&gt;
</pre>
</div></div>

<p>I will try to see if they work fine, and if they do, I will think about how to integrate them into the warnings plugin.</p>
</pre>
</details>

</ul>
</details>

<p>I have news about Eclipse CDT GCC error parser. I looked thouroughly into the source code and I found how GCC messages are parsed.<br/>
Actually, parsing is done with a generic Regexp parser dynamically configured with an XML file and a properties file.<br/>
There are several regular expressions involved in parsing GCC errors.</p>

<p>If you're interested see CDT Sources at: :pserver:anonymous@​dev.eclipse.org:2401/cvsroot/tools<br/>
in the directory: org.eclipse.cdt/all/org.eclipse.cdt.core<br/>
files: plugin.xml plugin.properties</p>

<p>Here are the regular expressions:</p>

<pre>
CDTGNUCErrorParser.name=CDT GNU C/C++ Error Parser
CDTGNUCErrorParser.regex.ReportedOnlyOnce=(.*?):(\\d+):(\\d+:)? .*\\(Each undeclared identifier is reported only once.*
CDTGNUCErrorParser.regex.ForEachFunctionItAppearsIn=(.*?):(\\d+):(\\d+:)? .*<span class="code-keyword">for</span> each function it appears in.\\).*
CDTGNUCErrorParser.regex.ReportedOnlyOncePerInputFile=(.*?):(\\d+):(\\d+:)? .*<span class="code-keyword">this</span> will be reported only once per input file.*
CDTGNUCErrorParser.regex.InstantiatedFromHere=(.*?):(\\d+):(\\d+:)?\\s*(.*instantiated from here.*)
CDTGNUCErrorParser.regex.GenericInfo=(.*?):(\\d+):(\\d+:)?\\s*(([Nn]ote)|(NOTE)|([Ii]nfo)|(INFO)): (.*)
CDTGNUCErrorParser.regex.ParseErrorBefore=(.*?):(\\d+):(\\d+:)? (parse error before.*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.ErrorUndeclared=(.*?):(\\d+):(\\d+:)? [Ee]rror: ([`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] undeclared .*)
CDTGNUCErrorParser.regex.ErrorConflictingTypesFor=(.*?):(\\d+):(\\d+:)? [Ee]rror: (conflicting types <span class="code-keyword">for</span> .*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.GenericError=(.*?):(\\d+):(\\d+:)?\\s*(([Ee]rror)|(ERROR)): (.*)
CDTGNUCErrorParser.regex.DefinedButNotUsed=(.*?):(\\d+):(\\d+:)? [Ww]arning: ([`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] defined but not used.*)
CDTGNUCErrorParser.regex.WarningConflictingTypesFor=(.*?):(\\d+):(\\d+:)? [Ww]arning: (conflicting types <span class="code-keyword">for</span> .*[`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNUCErrorParser.regex.GenericWarning=(.*?):(\\d+):(\\d+:)?\\s*(([Ww]arning)|(WARNING)): (.*)
CDTGNUCErrorParser.regex.OtherError=(.*?):(\\d+):(\\d+:)? (.*)

CDTGNULinkerErrorParser.name=CDT GNU Linker Error Parser
CDTGNULinkerErrorParser.regex.InFunction=(.*?):?(\\(\\.\\w+\\+.*\\))?:\\s*(In function [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>]:)
CDTGNULinkerErrorParser.regex.WarningDangerousFunction=(.*?):(\\d+):(\\d+:)? ([Ww]arning:)?\\s*(the use of [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>] is dangerous, better use [`<span class="code-quote">'<span class="code-quote">"](.*)['</span>"</span>].*)
CDTGNULinkerErrorParser.regex.TextWarning=(.*?):?\\(\\.\\w+\\+.*\\): [Ww]arning:? (.*)
CDTGNULinkerErrorParser.regex.TextError=(.*?):?\\(\\.\\w+\\+.*\\): (.*)
CDTGNULinkerErrorParser.regex.ldWarning=ld(\\.exe)?: [Ww]arning:? (.*)
CDTGNULinkerErrorParser.regex.ldError=ld(\\.exe)?: (.*)
</pre>
<p>And here is the piece of XML where they are declared:</p>

<pre>
   <extension
 id=<span class="code-quote">"GCCErrorParser"</span>
 name=<span class="code-quote">"%CDTGNUCErrorParser.name"</span>
 point=<span class="code-quote">"org.eclipse.cdt.core.ErrorParser"</span>>
      <errorparser
    class=<span class="code-quote">"org.eclipse.cdt.core.errorparsers.RegexErrorParser"</span>
    id=<span class="code-quote">"org.eclipse.cdt.core.GCCErrorParser"</span>
    name=<span class="code-quote">"%CDTGNUCErrorParser.name"</span>>
 <pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ReportedOnlyOnce<span class="code-quote">" severity="</span>Ignore"/>
 <pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ForEachFunctionItAppearsIn<span class="code-quote">" severity="</span>Ignore"/>
 <pattern description-expr=<span class="code-quote">"" eat-processed-line="</span><span class="code-keyword">true</span><span class="code-quote">" file-expr="</span><span class="code-quote">" line-expr="</span><span class="code-quote">" regex="</span>%CDTGNUCErrorParser.regex.ReportedOnlyOncePerInputFile<span class="code-quote">" severity="</span>Ignore"/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ErrorUndeclared"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ErrorConflictingTypesFor"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.ParseErrorBefore"</span> severity=<span class="code-quote">"Error"</span> variable-expr=<span class="code-quote">"$5"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.DefinedButNotUsed"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$5"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.WarningConflictingTypesFor"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$5"</span>/>
 <pattern description-expr=<span class="code-quote">"$5"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNULinkerErrorParser.regex.WarningDangerousFunction"</span> severity=<span class="code-quote">"Warning"</span> variable-expr=<span class="code-quote">"$6"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.InstantiatedFromHere"</span> severity=<span class="code-quote">"Info"</span>/>
 <pattern description-expr=<span class="code-quote">"$7"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericError"</span> severity=<span class="code-quote">"Error"</span>/>
 <pattern description-expr=<span class="code-quote">"$7"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericWarning"</span> severity=<span class="code-quote">"Warning"</span>/>
 <pattern description-expr=<span class="code-quote">"$9"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.GenericInfo"</span> severity=<span class="code-quote">"Info"</span>/>
 <pattern description-expr=<span class="code-quote">"$4"</span> eat-processed-line=<span class="code-quote">"<span class="code-keyword">true</span>"</span> file-expr=<span class="code-quote">"$1"</span> line-expr=<span class="code-quote">"$2"</span> regex=<span class="code-quote">"%CDTGNUCErrorParser.regex.OtherError"</span> severity=<span class="code-quote">"Error"</span>/>
      </errorparser>
   </extension>
</pre>

<p>I will try to see if they work fine, and if they do, I will think about how to integrate them into the warnings plugin.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=137153&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137153">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Did you made some progress on that issue? Do you need some help?</p>

</pre>
</details>

</ul>
</details>

<p>Did you made some progress on that issue? Do you need some help?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fchateau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=137227&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137227">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, I made some progress. I have something that works, and I've been testing it for a few weeks on my project and it seems satisfying. I just need to merge my modifications with the head revision and if it's fine I can commit (The admin granted me commit permission). I can send you a patch if you want to review my changes before. In fact, I was waiting for <a href="https://issues.jenkins.io/browse/JENKINS-5671" title="Warnings unit tests: parser matching parts of other parsers input" class="issue-link" data-issue-key="JENKINS-5671"><del>JENKINS-5671</del></a> resolution to commit, because otherwise my commit would have broken the tests and the way I solved this problem on my working copy was just dirty.</p>
</pre>
</details>

</ul>
</details>

<p>Yes, I made some progress. I have something that works, and I've been testing it for a few weeks on my project and it seems satisfying. I just need to merge my modifications with the head revision and if it's fine I can commit (The admin granted me commit permission). I can send you a patch if you want to review my changes before. In fact, I was waiting for <a href="https://issue-redirect.jenkins.io/issue/5671" title="Warnings unit tests: parser matching parts of other parsers input" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/5671">JENKINS-5671</a>"><del>JENKINS-5671</del></a> resolution to commit, because otherwise my commit would have broken the tests and the way I solved this problem on my working copy was just dirty.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=137229&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137229">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please go ahead and commit your changes directly. Thanks for improving the parser!</p>
</pre>
</details>

</ul>
</details>

<p>Please go ahead and commit your changes directly. Thanks for improving the parser!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=137245&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137245">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : fchateau
Path:
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/Gcc4CompilerParser.java
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/Gcc4LinkerParser.java
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/ParserRegistry.java
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/Gcc4CompilerParserTest.java
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/Gcc4LinkerParserTest.java
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/ParserRegistryTest.java
 trunk/hudson/plugins/warnings/src/test/resources/hudson/plugins/warnings/parser/gcc4.txt
 trunk/hudson/plugins/warnings/src/test/resources/hudson/plugins/warnings/parser/gcc4ld.txt
<a href="http://jenkins-ci.org/commit/29187" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/29187</a>
Log:
<a href="https://issues.jenkins.io/browse/JENKINS-4753" title="gcc warnings not parsed correctly" class="issue-link" data-issue-key="JENKINS-4753"><del>JENKINS-4753</del></a> NEW: added 2 new parsers: one for gcc 4.x compiler messages, and one for gcc 4.x linker messages. Added unit tests for these new parsers, and updated the tests of the parser registry.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : fchateau<br/>
Path:<br/>
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/Gcc4CompilerParser.java<br/>
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/Gcc4LinkerParser.java<br/>
 trunk/hudson/plugins/warnings/src/main/java/hudson/plugins/warnings/parser/ParserRegistry.java<br/>
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/Gcc4CompilerParserTest.java<br/>
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/Gcc4LinkerParserTest.java<br/>
 trunk/hudson/plugins/warnings/src/test/java/hudson/plugins/warnings/parser/ParserRegistryTest.java<br/>
 trunk/hudson/plugins/warnings/src/test/resources/hudson/plugins/warnings/parser/gcc4.txt<br/>
 trunk/hudson/plugins/warnings/src/test/resources/hudson/plugins/warnings/parser/gcc4ld.txt<br/>
<a href="http://jenkins-ci.org/commit/29187" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/29187</a><br/>
Log:<br/>
<a href="https://issue-redirect.jenkins.io/issue/4753" title="gcc warnings not parsed correctly" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/4753">JENKINS-4753</a>"><del>JENKINS-4753</del></a> NEW: added 2 new parsers: one for gcc 4.x compiler messages, and one for gcc 4.x linker messages. Added unit tests for these new parsers, and updated the tests of the parser registry.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torbent:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-4753?focusedId=137609&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-137609">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I checked this issue again today, and it does appear to be better now, but still a bit confusing.
To fchateau, who asked about compiler version:
I am running with g++ 4.4.1 and get the same output still as I did when I submitted this issue. Did you perhaps forget the "-Wsign-promo" option?</p>

<p>The current result of warning parsing (if I remember to only enable the gcc4 parser) is reported to be 4 warnings.
The summary says:</p>
<ul>
	<li>Total: 4</li>
	<li>High Proiority: 0</li>
	<li>Normal Priority: 6</li>
	<li>Low Priority: 0
If "Total" was meant to be the sum of the others, then something is wrong <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
</ul>


<p>Under "Details" in the "Warnings" tab, only one warning is listed:</p>
<blockquote><p>warnings.cc   6   Normal   gcc4   GCC4 Error</p></blockquote>
<p>with the mouseover text "passing 'Test' chooses 'int' over 'long unsigned int'"
That's quite fine, I think, except for the "GCC4 Error" part. It's only a warning.</p>

<p>In the "Details" tab, 4 warnings are listed, all in line 6. Their texts are:</p>
<blockquote>
<p>passing 'Test' chooses 'int' over 'long unsigned int'
passing 'Test' chooses 'int' over 'long int'
in call to 'std::basic_ostream&lt;_CharT, _Traits&gt;&amp; std::basic_ostream&lt;_CharT, _Traits&gt;::operator&lt;&lt;(int) <span class="error">&#91;with _CharT = char, _Traits = std::char_traits&#93;</span>'
passing 'Test' chooses 'int' over 'unsigned int'</p></blockquote>

<p>The "Normal" tab has the same contents as "Warnings", formatted differently.</p>

<p>The counts are confusing me a bit. Is this 1 or 4 warnings?
All in all, it's an improvement, but I'm not fully ready to close this issue yet.</p>
</pre>
</details>

</ul>
</details>

<p>I checked this issue again today, and it does appear to be better now, but still a bit confusing.<br/>
To fchateau, who asked about compiler version:<br/>
I am running with g++ 4.4.1 and get the same output still as I did when I submitted this issue. Did you perhaps forget the "-Wsign-promo" option?</p>

<p>The current result of warning parsing (if I remember to only enable the gcc4 parser) is reported to be 4 warnings.<br/>
The summary says:</p>
<ul>
	<li>Total: 4</li>
	<li>High Proiority: 0</li>
	<li>Normal Priority: 6</li>
	<li>Low Priority: 0<br/>
If "Total" was meant to be the sum of the others, then something is wrong <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
</ul>


<p>Under "Details" in the "Warnings" tab, only one warning is listed:</p>
<blockquote><p>warnings.cc   6   Normal   gcc4   GCC4 Error</p></blockquote>
<p>with the mouseover text "passing 'Test' chooses 'int' over 'long unsigned int'"<br/>
That's quite fine, I think, except for the "GCC4 Error" part. It's only a warning.</p>

<p>In the "Details" tab, 4 warnings are listed, all in line 6. Their texts are:</p>
<blockquote>
<p>passing 'Test' chooses 'int' over 'long unsigned int'<br/>
passing 'Test' chooses 'int' over 'long int'<br/>
in call to 'std::basic_ostream<_CharT, _Traits>& std::basic_ostream<_CharT, _Traits>::operator<<(int) <span class="error">&#91;with _CharT = char, _Traits = std::char_traits&#93;</span>'<br/>
passing 'Test' chooses 'int' over 'unsigned int'</p></blockquote>

<p>The "Normal" tab has the same contents as "Warnings", formatted differently.</p>

<p>The counts are confusing me a bit. Is this 1 or 4 warnings?<br/>
All in all, it's an improvement, but I'm not fully ready to close this issue yet.</p>
